### PR TITLE
SUBMARINE-657. The problem of creating a new user

### PR DESCRIPTION
**Pull request by charlie0220**
### What is this PR for?
1. Correct the bug while creating an user without entering birthday.
Now if users don't enter the birthday, the default birthday will be null.

2. Add md5 encryption for the password while creating new user.

### What type of PR is it?
[Bug Fix]

### Todos
* [ ] - Task

### What is the Jira issue?
[SUBMARINE-657](https://issues.apache.org/jira/projects/SUBMARINE/issues/SUBMARINE-657?filter=allopenissues)

### How should this be tested?
[Travis-CI](https://travis-ci.com/github/charlie0220/submarine)

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: submarine-workbench/workbench-web/src/app/pages/workbench/manager/user-drawer/user-drawer.component.ts [hunk 1]
```diff
@@ -153,6 +153,8 @@ export class UserDrawerComponent implements OnInit, OnChanges {
 
           if (formData.birthday) {
             formData.birthday = format(formData.birthday, 'yyyy-MM-dd HH:mm:ss');
+          }else{
```
Comment (pingsutw) [inline submarine-workbench/workbench-web/src/app/pages/workbench/manager/user-drawer/user-drawer.component.ts:156]:
```suggestion
          } else {
```

**Review by pingsutw — COMMENTED** [review]
Thanks @charlie0220 for the contribution.

File: submarine-workbench/workbench-web/src/app/services/user.service.ts [hunk 2]
```diff
@@ -96,6 +96,7 @@ export class UserService {
 
   createUser(sysUser: Partial<SysUser>): Observable<SysUser> {
     const apiUrl = this.baseApi.getRestApi('/sys/user/add');
+    sysUser['password'] = md5(sysUser.password);
```
Comment (xunliu) [inline submarine-workbench/workbench-web/src/app/services/user.service.ts:99]: The user password stored in the database is already encrypted.

File: submarine-workbench/workbench-web/src/app/pages/workbench/manager/user-drawer/user-drawer.component.ts [hunk 3]
```diff
@@ -153,6 +153,8 @@ export class UserDrawerComponent implements OnInit, OnChanges {
 
           if (formData.birthday) {
             formData.birthday = format(formData.birthday, 'yyyy-MM-dd HH:mm:ss');
+          } else {
+            formData.birthday = format(new Date(2000,0O0,0O1),'yyyy-MM-dd HH:mm:ss');
```
Comment (xunliu) [inline submarine-workbench/workbench-web/src/app/pages/workbench/manager/user-drawer/user-drawer.component.ts:157]: What time is `Date(2000,0O0,0O1)`?

File: submarine-workbench/workbench-web/src/app/services/user.service.ts (hunk 2 above)
Comment (charlie0220) [reply submarine-workbench/workbench-web/src/app/services/user.service.ts:99]: It seems like the password is not encrypted while creating a new user.
password

File: submarine-workbench/workbench-web/src/app/pages/workbench/manager/user-drawer/user-drawer.component.ts (hunk 3 above)
Comment (charlie0220) [reply submarine-workbench/workbench-web/src/app/pages/workbench/manager/user-drawer/user-drawer.component.ts]: I have updated this file. Now if user do not enter the birthday, the value stored into database will be null.

Comment (xunliu) [reply submarine-workbench/workbench-web/src/app/pages/workbench/manager/user-drawer/user-drawer.component.ts]: Okay, What we should do is to encrypt the password in the database.

**Review by lowc1012 — COMMENTED** [review]
@charlie0220 Thank you for your contribution.
I test this patch, and it works for me.

File: submarine-workbench/workbench-web/src/app/services/user.service.ts (hunk 2 above)
Comment (xunliu) [inline submarine-workbench/workbench-web/src/app/services/user.service.ts:99]: @charlie0220 When creating a user, you need to encrypt the user's password.
When the user logs in, the user’s password needs to be encrypted, and then corrected with the encrypted password in the database.

**Review by xunliu — APPROVED** [review]
LGTM

**Pull request closed without merging**